app/assets/javascripts/oxalis/view/right_menu_view.js

```diff
@@ -22,11 +23,12 @@ type Props = {
   viewMode: ModeType,
 };
 
-class RightMenuView extends React.PureComponent<Props> {
+class RightMenuView extends React.Component<Props> {
```

Comment:
Sorry if this is a stupid question, but why can't this component be pure now? :)

Reply:
Within the render function we access `Model.getSegmentationBinary` which is not pure since it doesn't depend on the props. In order to not lose updates, this cannot be a pure component and has to marked as such when connecting to the store :S After the model refactoring, this should be avoidable :)